src/batou_ext/python.py

```diff
@@ -217,26 +218,45 @@ def update(self):
                 return
 
             # add user env to DT_RPATH
-            self.__patchelf(["--add-rpath", directories], files_to_fix)
-            # drop everything from DT_RPATH except self.env_directory
+            # & drop everything from DT_RPATH except self.env_directory
             # and directories in the venv (to allow shared libraries from numpy
             # to load other shared libraries from numpy).
             self.__patchelf(
                 [
-                    "--shrink-rpath",
+                    "--add-rpath-and-shrink",
+                    directories,
                     "--allowed-rpath-prefixes",
                     f"{directories}:$ORIGIN",
                 ],
                 files_to_fix,
             )
 
     def __patchelf(self, args, paths):
+        patchelf_expr = shlex.quote(
+            dedent(
+                """
+        with import <nixpkgs> {}; {
+          patchelf = pkgs.patchelf-venv or patchelf.overrideAttrs ({ patches ? [], ... }: {
```

Comment:
The idea here is to have a `patchelf-venv` in the fc-nixos overlay to be able to migrate away from this inline Nix code at some point.
Not sure if that's reasonable (given we may get this upstreamed)?

Reply:
Maybe you put this comment *into* the file?

Reply:
Yeah, wanted to make sure we're talking about this during a review, will add a comment.
That said, I'm considering to throw this against upstream first and see if they're willing to accept such a patch. If so, I'd probably not do the fc-nixos dance and make sure that we'll get rid of the entire override eventually.